src/org/jitsi/impl/neomedia/rtp/translator/SimulcastController.java

```diff
@@ -147,6 +147,20 @@ public long getTargetSSRC()
      */
     public boolean accept(byte[] buf, int off, int len)
     {
+        int targetIndex = bitstreamController.getTargetIndex()
+            , currentIndex = bitstreamController.getCurrentIndex();
+
+        if (targetIndex < 0 && currentIndex > -1)
+        {
+            synchronized (this)
```

Comment:
No, unfortunately that wouldn't work. Suppose we remove the synchronized. And suppose that the filter thread has finished creating a new BitstreamController and is about to assign it to the instance field. Now, suppose another thread is calling SimulcastController.setTargetIndex which will call the BitstreamController.setTargetIndex of the old bitstream controller. The filter thread is left with a stale targetIndex and we don't want that.